pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -59,22 +107,36 @@ func (c *httpScrapeClient) Scrape(url string) (Stat, error) {
 		return emptyStat, fmt.Errorf("GET request for URL %q returned HTTP status %v", url, resp.StatusCode)
 	}
 	if resp.Header.Get("Content-Type") == network.ProtoAcceptContent {
-		return statFromProto(resp.Body)
+		return statFromProtoOptimized(resp.Body, resp.ContentLength)
 	}
 
 	return statFromPrometheus(resp.Body)
 }
 
-func statFromProto(body io.Reader) (Stat, error) {
+func statFromProtoOptimized(body io.Reader, l int64) (Stat, error) {
 	var stat Stat
-	bodyBytes, err := ioutil.ReadAll(body)
+	if l <= 0 {
+		return emptyStat, fmt.Errorf("no data received, data size unknown")
+	}
+	b, i := getDataBuffer(l)
+	defer func() {
+		pools[i].Put(b)
+	}()
+	var err error
+	var n int = 0
+	for n < int(l) && err == nil {
```

Comment:
Not using `Readful` as it requires to allocate an exact sized buffer.

Reply:
I guess the comment got under the wrong conversation (I blame github UI):

> I just wanted to get rid of the redundant checks because i already know the size of the buffer.

We're saving one or two `if` statements by manually inlining the code, and we're talking about code that's bottlenecked by networking overhead anyway. I'd much rather use the stdlibs implementation and not worry about small bugs here.

Reply:
Ok sure.